database/migrations/2020_01_12_000000_create_shoppingcart_table.php

```diff
@@ -11,20 +11,21 @@ class CreateShoppingcartTable extends Migration
      */
     public function up()
     {
-        Schema::create('shoppingcart', function (Blueprint $table) {
+        Schema::create('shopping_carts', function (Blueprint $table) {
+            $table->bigIncrements('id');
             $table->string('identifier');
```

Comment:
should this be indexed ?

Reply:
Yes, it probably should. Even though we're not joining on it (we're joining on the `id`), we still need to be able to locate a cart instance by this column. Good call.

Reply:
Oh, actually, the `identifier` column is part of the compound unique index, too, so an separate index is redundant. No change required, then.